crates/ruff_linter/src/rules/ruff/rules/function_call_in_dataclass_default.rs

```diff
@@ -156,3 +158,53 @@ fn any_annotated(class_body: &[Stmt]) -> bool {
         .iter()
         .any(|stmt| matches!(stmt, Stmt::AnnAssign(..)))
 }
+
+fn is_immutable_newtype_call(
+    func: &Expr,
+    semantic: &SemanticModel,
+    extend_immutable_calls: &[QualifiedName],
+) -> bool {
+    let Expr::Name(name) = func else {
+        return false;
+    };
+
+    let Some(binding) = semantic.only_binding(name).map(|id| semantic.binding(id)) else {
+        return false;
+    };
+
+    if !binding.kind.is_assignment() {
+        return false;
+    }
+
+    let Some(Stmt::Assign(StmtAssign { value, .. })) = binding.statement(semantic) else {
+        return false;
+    };
+
+    let Some(original_type) = newtype_base_type(value, semantic) else {
+        return false;
+    };
+
+    is_immutable_annotation(original_type, semantic, extend_immutable_calls)
+}
+
+fn newtype_base_type<'a>(expr: &'a Expr, semantic: &SemanticModel) -> Option<&'a Expr> {
+    let Expr::Call(ExprCall {
+        func, arguments, ..
+    }) = expr
+    else {
+        return None;
+    };
+
+    if !semantic.match_typing_expr(func, "NewType") {
+        return None;
+    }
+
+    if !arguments.keywords.is_empty() {
+        return None;
+    }
```

Comment:
Isn't the following also a valid form (using keyword arguments) ?
```py
from typing import NewType

UserId = NewType(name="UserId", tp=int)
```

Reply:
That's a good point.

I don't see any mention of whether `NewType` is required to be used with positional arguments either on the Python documentation nor in any of the PEPs. I think we should support them as I see usages of that on GitHub: https://github.com/search?q=NewType(name%3D+language:Python&type=code